doc/design/backup_operator.md

```diff
@@ -0,0 +1,47 @@
+# etcd Backup Operator Design Document
+
+## Problem
+
+Currently, etcd-operator supports various backup strategies such as PV, S3, and Azure Blob Store. However, those strategies are embedded into the etcd operator code itself. Support for new backup strategy are required to be pushed into etcd-operator codebase. As more backup strategies are added, the etcd-operator code base would be hard to maintain. In Addition, users may want to implement their backup strategy without confining to the current etcd-operator’s backup spec and may not want to dependent on etcd-operator codebase for backup.
+
+## Solution
+
+To give user the flexibility of handling etcd backup his/her own way and to decouple backup logic out of etcd operator, a standalone etcd-backup operator can be used to manage the backup.
+
+## A Proposed Architecture 
+
+In order to control backup of etcd clusters, a standalone etcd backup operator can be used to listen for a customized backup CRD:
+
+* Check if CRD exists.
+* Listen for new backup CR.
+* Manage etcd cluster backup basing on the incoming backup CR.
+
+CRD Definition:
+```yaml
+apiVersion: apiextentions.k8s.io/v1beta2
+kind: CustomResourceDefiniton
+metadata:
+  name: etcdbackups.etcd.database.coreos.com
+spec:
+  group: etcd.database.coreos.com
+  version: v1alpha1
+  Scope: Namespaced
+  names: 
+    kind: EtcdBackup
+    plural: etcdbackups
+```
+
+CR Definition:
+```yaml
+apiVersion: "etcd.database.coreos.com/v1alpha1"
+kind: "EtcdBackup"
+metadata:
+  name: example-etcd-cluster
+spec:
+  clusterName: example-etcd-cluster
+  storageType: <your-storage-type>
+  <your-storage-type>:
```

Comment:
If it is not adding any details here, probably just use `...` for everything from below.

Then start defining Go type for BackupSpec:
```Go
type BackupSpec struct {
}
... Status
```